run/idp-sql/cloud-sql.js

```diff
@@ -0,0 +1,178 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const Knex = require('knex');
+const { getCredConfig } = require('./secrets');
+const { logger } = require('./logging');
+
+const TABLE = process.env.TABLE || 'votes';
+let knex, credConfig;
+
+// Connection pooling config
+// See Cloud SQL sample https://github.com/GoogleCloudPlatform/nodejs-docs-samples/tree/master/cloud-sql/postgres/knex
+// to learn more about this configuration
+const config = {
+  pool: {
+    max: 5,
+    min: 5,
+    acquireTimeoutMillis: 60000,
+  },
+  createTimeoutMillis: 30000,
+  idleTimeoutMillis: 600000,
+  createRetryIntervalMillis: 200,
+}
```

Comment:
@kurtisvg is this recommended config for Serverless connection pools? I can't find a reference in the docs to remind myself.

Reply:
Unfortunately no magic recommendation based on environment - it's a function of many variables including things like load size and tolerability of latency.